py_zipkin/zipkin.py

```diff
@@ -463,16 +460,14 @@ def add_sa_binary_annotation(
             service_name=service_name,
             host=host,
         )
-        sa_binary_annotation = create_binary_annotation(
-            key=zipkin_core.SERVER_ADDR,
-            value=SERVER_ADDR_VAL,
-            annotation_type=zipkin_core.AnnotationType.BOOL,
-            host=sa_endpoint,
-        )
         if not self.logging_context:
-            self.sa_binary_annotations.append(sa_binary_annotation)
+            if self.sa_endpoint is not None:
```

Comment:
On second thought, I think it makes sense to at least handle the case where the IP address isn't set in the first annotation. 

I.e We can throw an error if the IP address is not None while resetting the SA annotation. This shouldn't complicate the logic too much

Reply:
My biggest worry with that is that it complicates the API and makes it hard to understand whether you can or cannot reset the SA endpoint.

Also, what would I do if the 2 endpoints have a different service_name or port?

Reply:
Yeah, I guess ideally if we did support this, we would rename the function to `merge_sa_annotations`. Maybe we could deprecate the `add_sa_binary_annotation` function (i.e print a warning) and recommend that they use the new one?

We could throw a similar error for endpoints that differ in service_name or port as well

Reply:
That'd mean adding a new function, they'd still need the old one to add the first sa annotation, no?

It's duable, the thing I'm not convinced of is that it's a good enough use case to be worth complicating the logic to support it. It feels to me they can just wait and set the sa annotation once they know the remote address. I don't see why they'd need to set it in 2 steps.

Reply:
This doesn't jive with the zipkinCore documentation:
```
* Indicates a server address ("sa") in a span. Most likely, there's only one.
* Multiple addresses are possible when a client is redirected, or fails to a
* different server ip or port.
```

Though I think it would probably be ok to restrict this (with a note in the documentation) if it makes it easier to move to v2

Reply:
@adriancole how's that represented in V2 spans?

Reply:
the zipkinCore docs need to be fixed. multiple attempts in one span is generally a bad idea, which invalidates the need for multiple sa. we shouldn't recommend it (i probably wrote that note during various attempts to reverse engineer rationale ;))

any merging can be done client side. there's no representation of multiple addresses in v2 either local or remote.